File: Assignment_1_2.py
#Implement binary search using divide and conquer
a = [1,2,3,4,5,6]
target = 5
left = 0
right = len(a)-1
while left<=right:
    
    print(a)
    mid = left+(right-left)//2 #middle index
    mid_e = a[mid] #middle element
    print(mid_e)
    if target == mid_e:
        print("Found")
        break
    elif target < mid_e:
        right = mid - 1
        
        
    else:
        left = mid + 1
else:
    print("target does not exist")    

File: Assignment_1_2.c
//implement binary search using divide and conquer
#include <stdio.h>

int main(){
    int a[] = {1,2,3,4,5,6};
    int target = 5;int left;int right;
    left = 0;
    int n = sizeof(a) / sizeof(a[0]);
    right = n-1;
    while (left<=right){
        int mid = left + (right - left)/2;
        int mid_e = a[mid];
        if (target == mid_e){
            printf("Found\n");
            break;
        }
        else if (target<mid_e){
            right = mid-1;
        }
        else
            left = mid+1;

    }
printf("Not found\n");
return 0;
}